File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <cstdio>
using namespace std;

// https://www.acmicpc.net/problem/2606

int map[101][101];
int visited[101];
int N;
int M;
int ans = 0;

void input() {
    scanf("%d%d", &N, &M);
    for (int i = 0; i < M; i++) {
        int a, b;
        scanf("%d%d", &a, &b);
        map[a][b] = 1;
        map[b][a] = 1;
    }
}

void dfs(int now) {
    for (int i = 1; i <= N; i++) {
        if (i == now) continue;
        if (visited[i]) continue;
        if (map[now][i] == 0) continue;
        visited[i] = 1;
        ans++;
        dfs(i);
    }
}

int main(void) {
    input();
    visited[1] = 1;
    dfs(1);
    printf("%d\n", ans);
}

File: 백준/Silver/2606. 바이러스/바이러스.py

V = int(input())
E = int(input())
graph = [[] for _ in range(V+1)]
for edge in range(E):
    st, en = map(int, input().split())
    graph[st].append(en)
    graph[en].append(st)
stk = [1]
visited = [0] * (V+1)
visited[1] = 1
ans = 0
while stk:
    cur = stk.pop()
    if cur != 1:
        ans += 1
    for vertex in graph[cur]:
        if visited[vertex]: continue
        visited[vertex] = 1
        stk.append(vertex)
print(ans)
